pkg/ccl/backupccl/backup_test.go

```diff
@@ -1499,6 +1499,153 @@ func TestBackupRestoreSystemJobsProgress(t *testing.T) {
 	checkInProgressBackupRestore(t, checkFraction, checkFraction)
 }
 
+// TestRestoreCheckpointing checks that progress is being persisted to the job record
+// when progress is made in the form of a span frontier.
+func TestRestoreCheckpointing(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+	defer jobs.TestingSetProgressThresholds()()
+
+	var allowResponse chan struct{}
+	params := base.TestClusterArgs{}
+	knobs := base.TestingKnobs{
+		DistSQL: &execinfra.TestingKnobs{
+			BackupRestoreTestingKnobs: &sql.BackupRestoreTestingKnobs{
+				RunAfterProcessingRestoreSpanEntry: func(_ context.Context) {
+					<-allowResponse
+				},
+			},
+		},
+		JobsTestingKnobs: jobs.NewTestingKnobsWithShortIntervals(),
+	}
+	testServerArgs := base.TestServerArgs{DisableDefaultTestTenant: true}
+	params.ServerArgs = testServerArgs
+	params.ServerArgs.Knobs = knobs
+
+	ctx := context.Background()
+	_, sqlDB, dir, cleanupFn := backupRestoreTestSetupWithParams(t, multiNode, 1,
+		InitManualReplication, params)
+	conn := sqlDB.DB.(*gosql.DB)
+	defer cleanupFn()
+
+	sqlDB.Exec(t, `CREATE DATABASE r1`)
+	// We create these tables to ensure there are enough spans to restore and that we have partial progress
+	// when stopping the job.
+	var numTables int
+	for char := 'a'; char <= 'g'; char++ {
+		tableName := "r1." + string(char)
+		sqlDB.Exec(t, fmt.Sprintf(`CREATE TABLE %s (id INT PRIMARY KEY, s STRING)`, tableName))
+		sqlDB.Exec(t, fmt.Sprintf(`INSERT INTO %s VALUES (1, 'x'),(2,'y')`, tableName))
+		numTables += 1
+	}
+	sqlDB.Exec(t, `BACKUP DATABASE r1 TO 'nodelocal://0/test-root'`)
+
+	restoreQuery := `RESTORE DATABASE r1 FROM 'nodelocal://0/test-root' WITH detached, new_db_name=r2`
+
+	backupTableID := sqlutils.QueryTableID(t, conn, "r1", "public", "a")
+
+	var jobID jobspb.JobID
+	// The do function on a restore stops more progress from being persisted to the job record
+	// after some progress is made.
+	do := func(query string, check inProgressChecker) {
+		t.Logf("checking query %q", query)
+
+		var totalExpectedResponses int
+		if strings.Contains(query, "RESTORE") {
```

Comment:
This looks copied from the old test. `if strings.Contains(query, "RESTORE")` will always be the case in this test. Please clean this do function up if you don't plan to use common code with the other test.